.github/workflows/integration-tests.yml

```diff
@@ -361,6 +361,12 @@ jobs:
           pip uninstall -y triton
           cd python
           pip install -v -e '.[tests]'
+          echo "Artificial erroring out"
+          exit 1
+      - name: Clean up after a failed build
+        if: '!success()'
```

Comment:
I think you can give the previous step an id like `amd-install-triton` and check `if: steps.amd-install-triton.outcome == 'failure'` here? See https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#steps-context.